### DISCUSSION TITLE
[Question]: Can you calculate company financial information using OCF?

### DISCUSSION THREAD

**Discussion by shulmanbrent-pave** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#337] [Q&A]
### Question:

Is it possible to calculate the following values from a fully complete OCF dataset? If so, how would you go about doing it? Hoping for instructions on what files to look at / objects to combine.

<img width="300" alt="image" src="https://user-images.githubusercontent.com/81606764/208052060-b41efe48-9385-4967-ba5f-01948aa4a7f0.png">

**Comment by JSv4** [comment]
Hey @shulmanbrent-pave, thanks for the question. The best place to open those is in the discussions tab. I'm going to convert this to a discussion. 

In the meantime, though, a few quick thoughts:

1. Price per share is available on the [Stock Class object](https://open-cap-table-coalition.github.io/Open-Cap-Format-OCF/schema_markdown/schema/objects/StockClass/) you will have to create for a new class/series of stock. You'll want the `price_per_share` field to get the original issue price here (though, if it varies for any reason from purchaser to purchaser, you'll need to look at the stock issuance events for each purchase).
2. If by valuation, you want to know the effective pre-money valuation at the time of a round, we don't currently have a field for that. We do have Valuation objects for proper 409A valuations, but I wonder if it might be worth adding the capability to store financing valuations in there too. It would be pretty trivial.
3. Since the architecture is event-based, you'll have to aggregate some of the values you want from the issuance events to the investors:
    1. Shares issued will have to be aggregated by looking over all of the [stock issuance events](https://open-cap-table-coalition.github.io/Open-Cap-Format-OCF/schema_markdown/schema/objects/transactions/issuance/StockIssuance/) for the round's authorized security. 
    2. Capital raised will likewise need to be aggregated from the issuances
4. Two of your fields don't have simple sources at the moment:
    1. the `fullyDilutedShares` value is actually not a standard value. Different people calculate fully diluted differently, and some instruments specify how to perform this calculation. The Coalition will be releasing its own proposed "default" calculation, and you could rely on that and use an OCF export, but it's not public yet. 
    2. The concept of a financing is not currently tracked in the cap table. I think it might be worth creating an Financing object and corresponding events (open and close events, for example). It might also be worth providing an optional field to link an issuance to a financing as well. All of that said, this is not something that is typically done in "old-fashioned" Excel cap tables. The financing is an arbitrary concept that is typically determined by who purchased stock via a given Stock Purchase Agreement, but, often, these will be amended to bring in new investors, so you can have multiple "open" and "close" events. It's typically the job of the company, its attorneys and/or its cap table provider to track which issuances are part of what financing (vs a follow-on or a secondary or something). @jacobyavis and @pjohnmeyer, what do you guys think of potentially adding a data structure to track the concepts of financings (since we're all doing it outside of the cap table anyway and it's very closely related to the information in the cap table)? I wouldn't propose we add it for our upcoming stable release, however. The Financing object could also link to non-409A valuation objects so we could track valuations from financings easily too.
